docs/js-demo/edit.js

```diff
@@ -195,44 +196,50 @@ myCode1Mirror.on("changes", onEditChanges);
 
 jsCode1Mirror.setSize(null,codeMirrorDefaultHeight);
 
-
-
 //checks or unchecks the eval button
 function changeEvalButton(bool) {
   $('#option-eval').prop('checked', bool);
   onEvalButtonChange();
 }
 
-//creates a gist from OCaml code
+//creates a hashed URL from OCaml code IF encoded hash is under 1,950 characters
+//otherwise creates a github gist
 $('#share').click(function (e) {
-  var state = $(this).button('loading');
-  var request =
-  {
-    "description": "BuckleScript Gist",
-    "public": true,
-    "files": {
-      "gist.ml": {
-        "content": myCode1Mirror.getValue()
+  var hashedCode = btoa(myCode1Mirror.getValue());
+  if (hashedCode.length <= 1950) {
+    var url = 'https://bloomberg.github.io/bucklescript/js-demo/#' + hashedCode;
```

Comment:
I might have misunderstood your second question, but the length check in the code above is to encode only if the concatenated url is below the 2,000 character limit (1,950 for encoded string + 50 for the original url) imposed by some browsers.